### PR TITLE
[M2-6472] Hide CTA's not intended for R1

### PR DESCRIPTION
**Pull request by ChaconC**
### 📝 Description

🔗 [Jira Ticket M2-6472](https://mindlogger.atlassian.net/browse/M2-6472)
🔗 [Jira Ticket M2-6496](https://mindlogger.atlassian.net/browse/M2-6496)

Changes include:

- Added `enableActivityFilterSort` feature flag
- Added `enableActivityAssign` feature flag
- Added `enableParticipantConnections` feature flag
- Removed unused `resetLDContext` missing from M2-6462

### 📸 Screenshots

| Before (Optional)                      | After                                 |
| -------------------------------------- | ------------------------------------- |
| CleanShot 2024-05-03 at 11 49 49 | CleanShot 2024-05-03 at 11 49 30 |

### 🪤 Peer Testing

> [!NOTE]
> These new feature flags will be **`True`** for the `DEV` environment to allow for continued development, make sure to manually disable the flag or change environments to test functionality.

- Navigate to the activities overview, the Filters, Sort and Add Activity items should be displayed. 
- Navigate to a Participants Details page and use the overflow menu, the "Assign Activity" menu item should be visible.
- Update `.env` with `REACT_APP_LAUNCHDARKLY_CLIENT_ID` to use the `TEST` or `PROD` ID (Check [Confluence](https://mindlogger.atlassian.net/wiki/spaces/MINDLOGGER1/pages/487227432/Using+LaunchDarkly+with+MindLogger#Keys) for IDs).
- Restart the app
- Navigate to the activities overview and observe the missing elements.

### CODE REVIEW

File: src/shared/types/featureFlags.ts [hunk 1]
```diff
@@ -5,6 +5,10 @@ import { LDFlagValue } from 'launchdarkly-react-client-sdk';
 export const FeatureFlagsKeys = {
   enableMultiInformant: 'enableMultiInformant',
   enableMultiInformantTakeNow: 'enableMultiInformantTakeNow',
+  // TODO: M2-6519 Activity Filter Sort flag cleanup
```
Comment (sultanofcardio) [inline src/shared/types/featureFlags.ts]:
```suggestion
  // TODO: https://mindlogger.atlassian.net/browse/M2-6519 Activity Filter Sort flag cleanup
```

File: src/shared/types/featureFlags.ts [hunk 2]
```diff
@@ -5,6 +5,10 @@ import { LDFlagValue } from 'launchdarkly-react-client-sdk';
...
+  enableActivityFilterSort: 'enableActivityFilterSort',
+  // TODO: M2-6518 Assign Activity flag cleanup
```
Comment (sultanofcardio) [inline src/shared/types/featureFlags.ts]:
```suggestion
  // TODO: https://mindlogger.atlassian.net/browse/M2-6518 Assign Activity flag cleanup
```

**Review by sultanofcardio — APPROVED** [review]
This seems to resolve most of [M2-6496](https://mindlogger.atlassian.net/browse/M2-6496) as well. Does it make sense to just use this PR for both tickets?

**Comment by ChaconC** [comment]
You're right @sultanofcardio, I didn't even think to check the participant details page but this covers all AC.

**Review by mbanting — APPROVED** [review]
Looks good, and tested on this PR's preview environment. Elements are hidden. 
Do the preview environments use LD's UAT environment?

**Comment by ChaconC** [comment]
> Looks good, and tested on this PR's preview environment. Elements are hidden. Do the preview environments use LD's UAT environment?

Yep @mbanting, the preview environment uses the TEST/UAT id

**Review by sultanofcardio — APPROVED** [review]

**Pull request merged**